patches/api/0421-Paper-Item-API.patch

```diff
@@ -0,0 +1,582 @@
++
++    // CrossbowMeta
++    PropertyKey<Boolean> CHARGED = PropertyKey.of("Charged", Boolean.class);
++    PropertyKey<List<ItemStack>> CHARGED_PROJECTILES = PropertyKey.of("ChargedProjectiles", TypeFactory.parameterizedClass(List.class, ItemStack.class));
++
++    // CompassMeta
++    PropertyKey<Position> LODESTONE_POS = PropertyKey.of("LodestonePos", Position.class);
++    PropertyKey<NamespacedKey> LODESTONE_DIMENSION = PropertyKey.of("LodestoneDimension", NamespacedKey.class);
++    PropertyKey<Boolean> LODESTONE_TRACKED = PropertyKey.of("LodestoneTracked", Boolean.class);
++
++    // BundleMeta
++    PropertyKey<List<ItemStack>> ITEMS = PropertyKey.of("Items", TypeFactory.parameterizedClass(List.class, ItemStack.class));
++
++    // BookMeta
++    PropertyKey<String> TITLE = PropertyKey.of("title", String.class);
++    PropertyKey<String> TITLE_FILTERED = PropertyKey.of("filtered_title", String.class);
++    PropertyKey<String> AUTHOR = PropertyKey.of("author", String.class);
++    PropertyKey<List<Component>> PAGES = PropertyKey.of("pages", TypeFactory.parameterizedClass(List.class, Component.class));
++    PropertyKey<List<Component>> PAGES_FILTERED = PropertyKey.of("filtered_pages", TypeFactory.parameterizedClass(List.class, Component.class));
++    PropertyKey<BookMeta.Generation> GENERATION = PropertyKey.of("generation", BookMeta.Generation.class);
++    PropertyKey<Boolean> RESOLVED = PropertyKey.of("resolved", Boolean.class);
++
++    // UnsignedBookMeta
++    PropertyKey<List<String>> PAGES_WRITTEN = PropertyKey.of("pages", TypeFactory.parameterizedClass(List.class, String.class));
++
++    // BlockStateMeta
++    PropertyKey<MutablePropertyView> BLOCK_ENTITY_TAG = PropertyKey.of("BlockEntityTag", MutablePropertyView.class);
++    // Some odd entity tags included in item meta
++    PropertyKey<NamespacedKey> NOTE_BLOCK_SOUND = PropertyKey.of("note_block_sound", NamespacedKey.class);
++
++    // AxolotlBucket
++    PropertyKey<Axolotl.Variant> VARIANT = PropertyKey.of("Variant", Axolotl.Variant.class);
++
++    // MusicInstrumentMeta
++    PropertyKey<MusicInstrument> INSTRUMENT = PropertyKey.of("instrument", MusicInstrument.class);
++
++    // BannerMeta
++    PropertyKey<DyeColor> SHIELD_BANNER_BASE = PropertyKey.of("Base", DyeColor.class);
++
++}
+diff --git a/src/main/java/io/papermc/paper/inventory/item/properties/StewEffect.java b/src/main/java/io/papermc/paper/inventory/item/properties/StewEffect.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..fd286c088caa098447ae857d341a09ea6a936f45
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/inventory/item/properties/StewEffect.java
+@@ -0,0 +1,26 @@
++package io.papermc.paper.inventory.item.properties;
++
++import org.bukkit.potion.PotionEffectType;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++/**
++ * Represents a serialized stew effect with a duration.
++ */
++public interface StewEffect {
```

Comment:
I think similarly to enchantments and attributes, it'd be best to expose a custom MultiMap implementation of stew effects